exec/csnex/Extensions.cs

```diff
@@ -0,0 +1,36 @@
+﻿namespace csnex
+{
+    internal static partial class CodeExtensions
+    {
+        public static int Compare(this byte[] self, byte[] rhs)
+        {
+            if (self == null && rhs == null) {
+                return 0;
+            } else if (self != null && rhs == null) {
+                if (self.Length == 0) {
+                    return 0;
+                }
+                return 1;
+            } else if (self == null && rhs != null) {
+                if (rhs.Length == 0) {
+                    return 0;
+                }
+                return -1;
+            }
+
+            int res = 0;
+            for (int i = 0; i < self.Length && i < rhs.Length; i++) {
```

Comment:
I don't think this works correctly as written. Compare two strings where one is a prefix of the other, like `car` and `carpet`. What should the result of this function be?

Reply:
Rewrote to support better logic.